### PR TITLE
enhance: add teams notifications for critical actions

### PR DESCRIPTION
**Pull request by sjschlapbach**

### CODE REVIEW

File: apps/auth/src/lib/util.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import axios from 'axios'
+
+export async function sendTeamsNotifications(scope: string, text: string) {
+  if (process.env.TEAMS_WEBHOOK_URL) {
+    return axios.post(process.env.TEAMS_WEBHOOK_URL, {
+      '@context': 'https://schema.org/extensions',
+      '@type': 'MessageCard',
+      themeColor: '0076D7',
+      title: `Migration: ${scope}`,
```
Comment (rschlaefli) [inline apps/auth/src/lib/util.ts]:
```suggestion
      title: scope,
```

File: packages/graphql/src/lib/util.ts [hunk 2]
```diff
@@ -38,4 +39,18 @@ export function formatDate(dateTime: Date) {
   }
 }
 
+export async function sendTeamsNotifications(scope: string, text: string) {
+  if (process.env.TEAMS_WEBHOOK_URL) {
+    return axios.post(process.env.TEAMS_WEBHOOK_URL, {
+      '@context': 'https://schema.org/extensions',
+      '@type': 'MessageCard',
+      themeColor: '0076D7',
+      title: `Migration: ${scope}`,
```
Comment (rschlaefli) [inline packages/graphql/src/lib/util.ts]:
```suggestion
      title: scope,
```

File: packages/graphql/src/services/sessions.ts [hunk 3]
```diff
@@ -879,171 +910,176 @@ export async function deactivateSessionBlock(
...
-    ctx,
-    sessionId,
-    sessionBlockId,
-    activeInstanceIds,
-  })
+    return updatedSession
+  } catch (error) {
+    await sendTeamsNotifications(
+      'graphql/deactivateSessionBlock',
+      `ERROR - failed to deactivate session block ${sessionBlockId} in session ${session.id} with active block ${session.activeBlockId}: ${error}`
```
Comment (rschlaefli) [inline packages/graphql/src/services/sessions.ts]: @sjschlapbach tested that this outputs something, i.e., error is not an object that cannot be printed? or should we use error.message or similar?

**Review by rschlaefli — CHANGES_REQUESTED** [review]

File: packages/graphql/src/services/sessions.ts (hunk 3 above)
Comment (sjschlapbach) [reply packages/graphql/src/services/sessions.ts]: Using error.message would probably be a good idea, will be fixed

**Review by rschlaefli — APPROVED** [review]

**Pull request merged**